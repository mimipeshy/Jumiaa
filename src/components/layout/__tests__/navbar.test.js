import React from 'react';
import ReactDOM from 'react-dom';
import {
  BrowserRouter as Router,
  Route,
} from 'react-router-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Navbar from '../Navbar';

afterEach(cleanup);

it('renders NavBar without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Router>
      <Route>
        <Navbar />
      </Route>
    </Router>,
    div,
  );
});

it('matches snapshot Navbar', () => {
  const nav = renderer.create(
    <Router>
      <Route>
        <Navbar />
      </Route>
    </Router>,
  ).toJSON();
  expect(nav).toMatchSnapshot();
});
