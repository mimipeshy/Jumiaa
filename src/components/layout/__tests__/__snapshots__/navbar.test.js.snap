// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`machets snapshot Navbar 1`] = `
<div>
  <nav
    className="navbar navbar-light bg-dark mb-5"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand text-white text-lg brand-text"
          href="/"
          onClick={[Function]}
        >
          MovieSeriesInfo
        </a>
      </div>
      <ul
        className="navbar-nav ml-auto text-light d-inline-block"
      >
        <li
          className="nav-item d-inline-block mr-4"
        >
          <i
            className="fab fa-imdb fa-5x"
            id="imdb-logo"
          />
        </li>
        <li
          className="nav-item d-inline-block mr-4"
        >
          <i
            className="fab fa-react fa-5x"
            id="react-logo"
          />
        </li>
      </ul>
    </div>
  </nav>
</div>
`;

exports[`matches snapshot Navbar 1`] = `
<div>
  <nav
    className="navbar navbar-light bg-dark mb-5"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand text-white text-lg brand-text"
          href="/"
          onClick={[Function]}
        >
          MovieSeriesInfo
        </a>
      </div>
      <ul
        className="navbar-nav ml-auto text-light d-inline-block"
      >
        <li
          className="nav-item d-inline-block mr-4"
        >
          <i
            className="fab fa-imdb fa-5x"
            id="imdb-logo"
          />
        </li>
        <li
          className="nav-item d-inline-block mr-4"
        >
          <i
            className="fab fa-react fa-5x"
            id="react-logo"
          />
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
